### PR TITLE
Fix queue wakeups (#2912)

### PR DESCRIPTION
**Pull request by edenhill**
..and some minor cleanups

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -69,6 +75,10 @@ librdkafka v1.7.0 is feature release:
    created partition objects, or partitions that were changing leaders, to
    not have their message queues purged. This could cause
    `abort_transaction()` to time out. This issue is now fixed.
+ * In certain high-thruput produce rate patterns the producing could stall for
```
Comment (mhowlett) [inline CHANGELOG.md:78]: thruput -> throughput

Comment (mhowlett) [reply CHANGELOG.md:78]: producer -> producing

Comment (mhowlett) [reply CHANGELOG.md:78]: "the producing" -> ", producing"

File: src/rdkafka_queue.h [hunk 2]
```diff
@@ -42,6 +42,13 @@
 
 TAILQ_HEAD(rd_kafka_op_tailq, rd_kafka_op_s);
 
+/**
+ * @struct Queue for rd_kafka_op_t*.
+ *
+ * @remark All readers of the queue must call rd_kafka_q_mark_served()
+ *         after reading the queue (while still holding the queue lock) to
+ *         clear the wakeup-sent flag.
```
Comment (mhowlett) [inline src/rdkafka_queue.h:50]: seems like something that should be encapsulated, but also ok everything considered

File: src/rdkafka_queue.h [hunk 3]
```diff
@@ -90,8 +97,10 @@ struct rd_kafka_q_io {
 	rd_socket_t fd;
 	void  *payload;
 	size_t size;
-        rd_ts_t ts_rate;  /**< How often the IO wakeup may be performed (us) */
-        rd_ts_t ts_last;  /**< Last IO wakeup */
+        rd_bool_t sent; /**< Wake-up has been sent.
+                         *   This field is reset to false by the queue
+                         *   reader, allowing a new wake-up to be sent by a
+                         *   sub-sequent writer. */
```
Comment (mhowlett) [inline src/rdkafka_queue.h:103]: subsequent

**Review by mhowlett — APPROVED** [review]

File: src/rdkafka_queue.h (hunk 2 above)
Comment (edenhill) [reply src/rdkafka_queue.h:50]: It is encapsulated in the implementation (queue.c and queue.h)

**Pull request merged**